### DISCUSSION TITLE
Are the files that are "+" attached to single prompts stored or discarded after use? (not talking about the ones loaded in from the document library with #)

### DISCUSSION THREAD

**Discussion by odonnja** [open-webui/open-webui#1614] [Q&A]
I can see when I upload a document to the document library that it is stored in "uploads" in the open-webui volume in Docker, but I don't see some of the files that are attached to individual prompts in that folder. Are the files that are attached to individual prompts (by clicking the "+" button next to the prompt) stored there (or somewhere else) forever, or are they deleted when a chat is deleted? I feel like they should be deleted as part of the chat deletion process, but I don't know if they are and would like to know how all of that works if anyone knows.

**Comment by Classic298** [comment]
My opinion also is they should get deleted. Not sure why they are not. This is a problem

**Comment by dnesting** [comment]
They appear to be stored in a `webui.db` SQLite database with the conversation, and appear to be deleted when the conversation is deleted.

**Reply by Classic298** [reply]
No they are stored in chroma db. I think in the latest release the file itself is finally getting deleted when deleting a chat but the embeddings in the chroma db still do not get deleted, leading to an ever growing chroma db

**Reply by Classic298** [reply]
The only "solution" to this for now is this: https://github.com/open-webui/open-webui/discussions/5558#discussioncomment-11412332

Even the "reset vector storage" button in the Admin Panel **_does not work_**

**Comment by jehutywong** [comment]
in version 0.5.20, the file will not be deleted in all the webui.db, vector_db (my case elasticsearch) and data/uploads.

**Reply by Classic298** [reply]
sorry, is that a no(t) or a no(w) ? :D

**Reply by jehutywong** [reply]
it is a not, sorry about that
